datafusion/core/tests/sql/timestamp.rs

```diff
@@ -1411,14 +1411,14 @@ async fn cast_timestamp_before_1970() -> Result<()> {
 
     assert_batches_eq!(expected, &actual);
 
-    let sql = "select cast('1969-01-01T00:00:00.1Z' as timestamp);";
+    let sql = "select cast('1969-01-01T00:00:00.100Z' as timestamp);";
```

Comment:
what happens without this change?

Reply:
An error, it was a mis-reading of the RFC spec, I thought it restricted to multiples of 3 for subsecond precision, I'm working on a fix

Reply:
https://github.com/apache/arrow-rs/pull/3858 I think is related

Reply:
I would tend to think this regression should block the upgrade of datafusion, but would defer to others. 

maybe we can mark this PR as "ready for review" to get some other opinions 🤔